app/src/lib/stores/git-store.ts

```diff
@@ -145,11 +142,12 @@ export class GitStore extends BaseStore {
 
   private _stashEntryCount = 0
 
-  public constructor(repository: Repository, shell: IAppShell) {
+  public constructor(
+    private readonly repository: Repository,
+    private readonly shell: IAppShell,
+    private readonly statsStore: StatsStore
```

Comment:
I've added the `statsStore` as a parameter to the `gitStore`. This has caused a bunch of changes in tests but couldn't find an easier way to record the added tags (an alternative would have been to recalculate the difference between the previous state and the new state on the `appStore`, but I prefered to keep that logic on a single point.

Reply:
ugh. yeah i think that's the best solution for this without refactoring the entire interface for the stats store.

the only other place i could think of would be [`AppStore.OnGitStoreUpdated`](https://github.com/desktop/desktop/blob/1f15e696163ecad0f3e5a797492317d6d5977d8f/app/src/lib/stores/app-store.ts#L744), but it would involve a whole comparison iteration thing and we don't do any stats updating there already, so 🚢 it.